quickwit/quickwit-serve/src/log_level_handler.rs

```diff
@@ -0,0 +1,60 @@
+// Copyright (C) 2024 Quickwit, Inc.
+//
+// Quickwit is offered under the AGPL v3.0 and as commercial software.
+// For commercial licensing, contact us at hello@quickwit.io.
+//
+// AGPL:
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+use hyper::StatusCode;
+use serde::Deserialize;
+use tracing::{error, info};
+use warp::{Filter, Rejection};
+
+use crate::{with_arg, EnvFilterReloadFn};
+
+#[derive(Deserialize)]
+struct EnvFilter {
+    filter: String,
+}
+
+#[utoipa::path(get, tag = "Log level", path = "/log_level")]
+pub fn log_level_handler(
+    env_filter_reload_fn: EnvFilterReloadFn,
+) -> impl warp::Filter<Extract = (impl warp::Reply,), Error = Rejection> + Clone {
+    warp::path("log_level")
+        .and(warp::path::end())
+        .and(with_arg(env_filter_reload_fn))
```

Comment:
there is no http method selector. this also works with `POST`, `PUT`, `DELETE`, `INVALID`...
(nonblocking)